sources/packages/backend/apps/api/src/route-controllers/institution/_tests_/e2e/institution.aest.controller.getInstitutionDetailById.e2e-spec.ts

```diff
@@ -0,0 +1,103 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  AESTGroups,
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAESTToken,
+} from "../../../../testHelpers";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  createFakeInstitution,
+} from "@sims/test-utils";
+import { InstitutionDetailAPIOutDTO } from "../../models/institution.dto";
+import {
+  INSTITUTION_TYPE_BC_PRIVATE,
+  INSTITUTION_TYPE_BC_PUBLIC,
+} from "@sims/sims-db/constant";
+import { getISODateOnlyString } from "@sims/utilities";
+
+describe("InstitutionAESTController(e2e)-getInstitutionDetailById", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+  });
+
+  it("Should return institution details when an institution with given institution id exist.", async () => {
+    // Arrange
+    const institution = await db.institution.save(createFakeInstitution());
+    const mailingAddress = institution.institutionAddress.mailingAddress;
+    const institutionType = await db.institutionType.findOne({
+      select: { id: true, name: true },
```

Comment:
We can skip fetching the `id` here and directly assign `institution.institutionType.id` in line 51.

Reply:
why, it is used in 
![image](https://github.com/bcgov/SIMS/assets/77353155/b0c4f954-823c-4cbb-8f9a-5d1c28c6114e)

Reply:
Yes we can directly use `institution.institutionType.id` but the transient object(which is not retrieved from database) does not have the institution type name at line 52. Hence I added the fetch here.